### PR TITLE
fix: Allow single scoring metric in estimator reports

### PR DESCRIPTION
**Pull request by mathieulngt**
### Summary  
This PR allows `metrics.report_metrics` to accept a single scoring metric and a single scoring name, whereas it previously only supported lists of scoring metrics and names.

### Changes  
- Updated the types of `scoring` and `scoring_names` to accept both single elements and lists  
- Automatically wrap single `scoring` and `scoring_names` inputs into one-element lists

### Motivation  
It is more natural and user-friendly to allow single values instead of requiring one-element lists.  
See the original issue: [#1518](#1518).

### CODE REVIEW

File: skore/src/skore/sklearn/_estimator/metrics_accessor.py [hunk 1]
```diff
@@ -63,8 +63,15 @@ def report_metrics(
...
-        scoring: Optional[list[Union[str, Callable, SKLearnScorer]]] = None,
-        scoring_names: Optional[list[Union[str, None]]] = None,
+        scoring: Optional[
+            Union[
+                str,
+                Callable[..., object],
+                SKLearnScorer,
+                list[Union[str, Callable[..., object], SKLearnScorer]],
+            ]
+        ] = None,
```
Comment (auguste-probabl) [inline skore/src/skore/sklearn/_estimator/metrics_accessor.py:73]: Define a type to make this more readable, e.g.
```suggestion
        scoring: Optional[Union[Scoring, list[Scoring]]] = None,
```

**Comment by glemaitre** [comment]
I resolved the conflict and make sure that we support this feature across reporters with additional tests.

@auguste-probabl If you want to have another look.

**Review by glemaitre — DISMISSED** [review]

File: skore/src/skore/sklearn/_cross_validation/metrics_accessor.py [hunk 2]
```diff
@@ -54,8 +55,8 @@ def report_metrics(
         data_source: DataSource = "test",
         X: Optional[ArrayLike] = None,
         y: Optional[ArrayLike] = None,
-        scoring: Optional[Union[list[str], Callable, SKLearnScorer]] = None,
-        scoring_names: Optional[list[str]] = None,
+        scoring: Optional[Union[Scoring, list[Scoring]]] = None,
+        scoring_names: Optional[Union[str, list[Union[str, None]]]] = None,
```
Comment (auguste-probabl) [inline skore/src/skore/sklearn/_cross_validation/metrics_accessor.py:59]:
```suggestion
        ScoringName = Union[str, None]
        scoring_names: Union[ScoringName, list[ScoringName]] = None,
```

Comment (auguste-probabl) [reply skore/src/skore/sklearn/_cross_validation/metrics_accessor.py:59]: Can you update the docs for `scoring_names`?

File: skore/src/skore/sklearn/_estimator/metrics_accessor.py [hunk 3]
```diff
@@ -30,6 +30,7 @@
 from skore.utils._index import flatten_multi_index
 
 DataSource = Literal["test", "train", "X_y"]
+Scoring = Union[str, Callable[..., object], SKLearnScorer]
```
Comment (auguste-probabl) [inline skore/src/skore/sklearn/_estimator/metrics_accessor.py:33]: Move it to `skore.sklearn.types` maybe

**Comment by thomass-dev** [comment]
Please update your PR with main, so that the `pytest` workflow status will be reported.

**Review by auguste-probabl — APPROVED** [review]
